AGENTS.md

```diff
@@ -0,0 +1,251 @@
+# AGENTS.md
+
+This file provides instructions for autonomous agents working in the operator-controller repository.
+
+## Repository Context
```

Comment:
But before we add it, what info here is valid only for upstream?
If we add this, will all be valid for downstream? I do not think so,
So, how will we have information for upstream and another for downstream?

@joelanford @tmshort have you ideas?
What about we ignore those files in the sync/bumper?
Do we want to have the same for downstream? Could we have a version there that is not removed by the bumper as well?